src/components/views/right_panel/UserInfoSharedRooms.tsx

```diff
@@ -0,0 +1,195 @@
+import SpecPermalinkConstructor from '../../../utils/permalinks/SpecPermalinkConstructor';
+import { mostRecentActivityFirst } from '../../../RoomListSorter';
+
+interface IProps {
+    userId: string;
+    compact: boolean;
+}
+
+interface IState {
+    roomIds?: [];
+    error: boolean;
+    showAll: boolean;
+}
+
+const LIMITED_VIEW_SHOW_COUNT = 3;
+
+export default class UserInfoSharedRooms extends React.PureComponent<IProps, IState> {
+
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            error: false,
+            showAll: false,
+        };
+    }
+
+    componentDidMount() {
+        return this.componentDidUpdate();
+    }
+
+    async componentDidUpdate(prevProps?: IProps) {
+        if (prevProps && prevProps.userId === this.props.userId) {
+            // Nothing to update.
+            return;
+        }
+
+        // Reset because this is a new user
+        this.setState({
+            error: false,
+            showAll: false,
+            roomIds: undefined,
+        });
+
+        try {
+            const roomIds = await MatrixClientPeg.get()._unstable_getSharedRooms(this.props.userId);
+            this.setState({
+                roomIds,
+            });
+        } catch (ex) {
+            console.log(`Failed to get shared rooms for ${this.props.userId}`, ex);
+            this.setState({ error: true });
+        }
+    }
+
+    private onRoomTileClick(roomId) {
+        dis.dispatch({
+            action: 'view_room',
+            show_room_tile: true, // to make sure the room gets scrolled into view
+            room_id: roomId,
+        });
+    }
+
+    private onShowMoreClick() {
+        console.log("Showing more");
+        this.setState({
+            showAll: true,
+        });
+    }
+
+    private renderRoomTile(room) {
+        // If the room cannot be found, hide it.
+        if (!room) {
+            return null;
+        }
+
+        // If the room has been upgraded, hide it.
+        const tombstone = room.currentState.getStateEvents("m.room.tombstone", "");
+        if (tombstone) {
+            return null;
+        }
+
+        if (this.props.compact) {
+            // XXX: This is inefficent as we only render COMPACT_VIEW_SHOW_COUNT rooms at a time, the other pills are wasted.
+            const alias = room.getCanonicalAlias();
+            if (!alias) {
+                // Without an alias we get ugly room_ids, hide it.
+                return null;
+            }
+            return <a href={`#/room/${alias}`}><Pill
+                key={room.roomId}
+                type={Pill.TYPE_ROOM_MENTION}
+                room={room}
+                url={new SpecPermalinkConstructor().forRoom(alias)}
+                inMessage={false}
+                shouldShowPillAvatar={true}
+                isSelected={false}
+            /></a>;
+        }
+
+        return <li key={room.roomId}>
+            <RoomTile
```

Comment:
just fyi that this component is disappearing in the new room list. Unfortunately the new room tile isn't meant to be used in this way either, so if product does approve this then it'll need to go through design review and get a custom implementation.

Reply:
This now has a `UserInfoRoomTile`